elastic/apm-server
# sampling: match policies in the given order

## Motivation/summary

Ensure tail-based sampling policies are matched in the order they are specified, regardless of the criteria.

## Checklist


## How to test these changes

1. Run APM Server with the following config:
```
apm-server:
  sampling:
    tail:
      enabled: true
      interval: 10s
      policies:
      - trace.name: no_spans
        sample_rate: 0.5
      - service.name: goagent
        sample_rate: 0.1
```
2. Send 1000 transactions with `transaction.name: no_spans`, and `service.name: goagent`.
3. 500 (50%) traces should be sampled.

## Related issues

Fixes https://github.com/elastic/apm-server/issues/4616
